File: models/household.js
module.exports = (sequelize, DataTypes) => {
  const Household = sequelize.define('Household', {
    id: {
      type: DataTypes.INTEGER,
      primaryKey: true,
      autoIncrement: true
    },
    name: {
      type: DataTypes.STRING(100),
      allowNull: false
    },
    description: {
      type: DataTypes.STRING(255),
      allowNull: true
    },
    financialMonthStartDay: {
      type: DataTypes.INTEGER,
      allowNull: false,
      defaultValue: 10,
      validate: { min: 1, max: 28 }
    }
  }, {
    timestamps: true,
    tableName: 'households'
  });

  return Household;
};

File: backend_go/internal/models/household.go
package models

import "time"

// Household is the top-level multi-tenant entity.
// Sequelize model: Household — table: households
type Household struct {
	ID          uint       `gorm:"primaryKey;autoIncrement" json:"id"`
	Name        string     `gorm:"type:varchar(255);not null" json:"name"`
	// *string means nullable VARCHAR — pointer semantics: nil = NULL in DB.
	// In Go, the zero value of string is "" (empty string), which is NOT the same
	// as NULL. Use *string when you need to distinguish between "no value" and "".
	Description *string    `gorm:"type:text" json:"description"`
	CreatedAt   time.Time  `json:"createdAt"`
	UpdatedAt   time.Time  `json:"updatedAt"`

	// HasMany associations (not DB columns — populated by GORM Preload)
	Members          []HouseholdMember        `gorm:"foreignKey:HouseholdID" json:"HouseholdMembers,omitempty"`
	ExpenseCategories []ExpenseCategory        `gorm:"foreignKey:HouseholdID" json:"ExpenseCategories,omitempty"`
	IncomeCategories  []IncomeCategory         `gorm:"foreignKey:HouseholdID" json:"IncomeCategories,omitempty"`
	Expenses          []Expense                `gorm:"foreignKey:HouseholdID" json:"-"`
	Incomes           []Income                 `gorm:"foreignKey:HouseholdID" json:"-"`
	Notes             []Note                   `gorm:"foreignKey:HouseholdID" json:"-"`
	Assets            []Asset                  `gorm:"foreignKey:HouseholdID" json:"-"`
	BudgetOverrides   []CategoryBudgetOverride `gorm:"foreignKey:HouseholdID" json:"-"`
}

func (Household) TableName() string { return "households" }
